File: app/src/main/java/one/mixin/android/widget/linktext/AutoLinkMode.java
package one.mixin.android.widget.linktext;


public enum AutoLinkMode {

    MODE_HASHTAG("Hashtag"),
    MODE_MENTION("Mention"),
    MODE_URL("Url"),
    MODE_PHONE("Phone"),
    MODE_EMAIL("Email"),
    MODE_BOT("BOT"),
    MODE_CUSTOM("Custom");

    private String name;

    AutoLinkMode(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: app/src/main/java/one/mixin/android/widget/linktext/AutoLinkMode.kt
package one.mixin.android.widget.linktext

enum class AutoLinkMode(name: String) {
    MODE_HASHTAG("Hashtag"),
    MODE_MENTION("Mention"),
    MODE_URL("Url"),
    MODE_PHONE("Phone"),
    MODE_EMAIL("Email"),
    MODE_BOT("Bot"),
    MODE_CUSTOM("Custom"),
    MODE_MARKDOWN_BOLD("Bold"),
    MODE_MARKDOWN_ITALIC("Slant"),
    MODE_MARKDOWN_STRIKETHROUGH("Strikethrough"),
    MODE_MARKDOWN_INLINE("inline");

    override fun toString(): String {
        return name
    }
}
